File: java-gradle/src/main/java/cz/petrfaltus/sta_ter_cont_ac_restclient/Web.java
package cz.petrfaltus.sta_ter_cont_ac_restclient;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStream;

import java.net.HttpURLConnection;
import java.net.URL;

public class Web {
    public static final String URL_ADDRESS = "http://api.petrfaltus.net/country_code_prefix/json/1.0";
    public static final String CHARSET = "UTF-8";
    public static final String USER_AGENT = "Petr Faltus Java State, territory, continent and area code REST client";

    public static String request(String requestBody) {
        String output = Const.EMPTY_STRING; // the default output value
        HttpURLConnection connection = null;

        try {
            URL url = new URL(URL_ADDRESS);

            connection = (HttpURLConnection) url.openConnection();
            connection.setUseCaches(false);
            connection.setDoOutput(true);
            connection.addRequestProperty("User-Agent", USER_AGENT);
            connection.setRequestMethod("POST");

            OutputStream os = connection.getOutputStream();
            byte[] requestBodyBytes = requestBody.getBytes(CHARSET);
            os.write(requestBodyBytes);

            InputStream is = connection.getInputStream();
            InputStreamReader isr = new InputStreamReader(is, CHARSET);
            BufferedReader br = new BufferedReader(isr);

            String line;
            while ((line = br.readLine()) != null) {
                output += line ;
                output += System.lineSeparator();
            }

            br.close();
        } catch (Exception e) {
            output = null; // indicates the web request failed
        } finally {
            if (connection != null) {
                connection.disconnect();
            }
        }

        return output;
    }
}

File: csharp/Web.cs
using System.Net;
using System.Text;

namespace StaTerContAcRestClient
{
    public class Web
    {
        private static readonly string URL_ADDRESS = "http://api.petrfaltus.net/country_code_prefix/json/1.0";
        private static readonly Encoding encoding = Encoding.UTF8;
        private static readonly string USER_AGENT = "Petr Faltus C# State, territory, continent and area code REST client";

        private static WebClient client = null;

        public static string Request(string input)
        {
            if (client == null)
            {
                client = new WebClient();
                client.Encoding = encoding;
                client.Headers.Add("user-agent", USER_AGENT);
            }

            string output = client.UploadString(URL_ADDRESS, input);

            return output;
        }
    }
}
